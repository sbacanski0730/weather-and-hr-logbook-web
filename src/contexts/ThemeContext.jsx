import { createContext, useContext, useEffect, useState } from 'react';

const ThemeContextNew = createContext();

// eslint-disable-next-line no-unused-vars, react/prop-types
export const ThemeProvider = ({ children }) => {
    const getTheme = () => localStorage.getItem('theme');
    const [themeMode, setThemeMode] = useState(getTheme() || 'light');

    const toggleTheme = () => {
        if (themeMode === 'light') setThemeMode('dark');
        if (themeMode === 'dark') setThemeMode('light');
    };

    useEffect(() => {
        const mode = localStorage.getItem('theme');
        if (mode) {
            if (mode.match(/light/i) || mode.match(/dark/i)) setThemeMode(mode.toLowerCase());
        } else {
            setThemeMode('light');
        }
    }, []);

    useEffect(() => {
        localStorage.setItem('theme', themeMode);
    }, [themeMode]);

    return (
        <>
            <ThemeContextNew.Provider value={{ themeMode, toggleTheme }}>{children}</ThemeContextNew.Provider>
        </>
    );
};

export default ThemeContextNew;

export const useTheme = () => {
    const theme = useContext(ThemeContextNew);
    if (!theme) throw Error('useTheme need to be used inside ThemeContextNew');
    return theme;
};
