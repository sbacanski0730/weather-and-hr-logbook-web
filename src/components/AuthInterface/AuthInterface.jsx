import React from 'react';
import { Link as ReactLink, useLocation } from 'react-router-dom';
import AppBar from '@mui/material/AppBar';
import Typography from '@mui/material/Typography';
import Stack from '@mui/material/Stack';
import Link from '@mui/material/Link';

import DarkModeSharpIcon from '@mui/icons-material/DarkModeSharp';
import LightModeIcon from '@mui/icons-material/LightMode';

import CustomTopbarIconButton from '../styled-components/CustomTopbarIconButton.jsx';
import CustomTopbarButton from '../styled-components/CustomTopbarButton.jsx';

import { APP_ROUTES } from '../../utils/constants';

import { useTheme } from '../../contexts/ThemeContext.jsx';

const AuthInterface = () => {
    // const { themeMode, toggleTheme } = useContext(ThemeContext);
    const { pathname } = useLocation();
    const { themeMode, toggleTheme } = useTheme();

    return (
        <AppBar
            position="fixed"
            sx={{
                backgroundColor: 'primary.main',
                display: 'flex',
                flexDirection: 'row',
                justifyContent: 'flex-end',
                alignItems: 'center',
                py: '3px',
                px: 1,
            }}
        >
            <Stack direction="row" spacing={1}>
                <CustomTopbarIconButton
                    onClick={toggleTheme}
                    size="small"
                    sx={{
                        px: '1px',
                        py: '1px',
                    }}
                >
                    {themeMode === 'dark' ? (
                        <LightModeIcon data-testid="light-mode-icon" />
                    ) : (
                        <DarkModeSharpIcon data-testid="dark-mode-icon" />
                    )}
                </CustomTopbarIconButton>
                <CustomTopbarButton sx={{ my: '1px', mx: '2px' }}>
                    {pathname.includes('info') ? (
                        <Link
                            to={APP_ROUTES.LOGIN}
                            component={ReactLink}
                            sx={{ color: 'inherited' }}
                            underline="none"
                            data-testid="custom-login-link"
                        >
                            <Typography variant="h5">LOGIN</Typography>
                        </Link>
                    ) : (
                        <Link
                            to={APP_ROUTES.INFO}
                            component={ReactLink}
                            sx={{ color: 'inherited' }}
                            underline="none"
                            data-testid="custom-info-link"
                        >
                            <Typography variant="h5">INFO</Typography>
                        </Link>
                    )}
                </CustomTopbarButton>
            </Stack>
        </AppBar>
    );
};

export default AuthInterface;
