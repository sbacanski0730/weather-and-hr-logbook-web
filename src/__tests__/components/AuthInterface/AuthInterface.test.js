import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import AuthInterface from '../../../components/AuthInterface/AuthInterface.jsx';
import { ThemeProvider } from '../../../contexts/ThemeContext.jsx';

describe('AuthInterface component', () => {
    it('Checks does the component has INFO element', async () => {
        render(
            <MemoryRouter initialEntries={[{ pathname: '/login' }]}>
                <ThemeProvider>
                    <AuthInterface />
                </ThemeProvider>
            </MemoryRouter>
        );
        const e = await screen.findByText(/info/i);
        expect(e).toBeInTheDocument();
    });

    // ====================
    it('Checks does the component has Login element', async () => {
        render(
            <MemoryRouter initialEntries={[{ pathname: '/info' }]}>
                <ThemeProvider>
                    <AuthInterface />
                </ThemeProvider>
            </MemoryRouter>
        );
        const e = await screen.findByText(/login/i);
        expect(e).toBeInTheDocument();
    });

    // ====================
    it('Checks does a proper icons was rendered in component', async () => {
        render(
            <MemoryRouter initialEntries={[{ pathname: '/login' }]}>
                <ThemeProvider>
                    <AuthInterface />
                </ThemeProvider>
            </MemoryRouter>
        );
        const icon = screen.getByTestId('dark-mode-icon');
        expect(icon).toBeInTheDocument();
    });
});
