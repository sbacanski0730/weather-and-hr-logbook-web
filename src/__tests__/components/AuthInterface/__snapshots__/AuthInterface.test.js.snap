// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthInterface component Component snapshot test 1`] = `
<header
  className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-1ktv3l4-MuiPaper-root-MuiAppBar-root"
>
  <div
    className="MuiStack-root css-w4z10b-MuiStack-root"
  >
    <button
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-j6pqhf-MuiButtonBase-root-MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="dark-mode-icon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 3c-4.97 0-9 4.03-9 9s4.03 9 9 9 9-4.03 9-9c0-.46-.04-.92-.1-1.36-.98 1.37-2.58 2.26-4.4 2.26-2.98 0-5.4-2.42-5.4-5.4 0-1.81.89-3.42 2.26-4.4-.44-.06-.9-.1-1.36-.1z"
        />
      </svg>
    </button>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-11pcvcz-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <a
        className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1pmsi6l-MuiTypography-root-MuiLink-root"
        data-testid="custom-info-link"
        href="/info"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
      >
        <h5
          className="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
        >
          INFO
        </h5>
      </a>
    </button>
  </div>
</header>
`;
